lib/utils/expandObjectTokens.js

```diff
@@ -178,11 +178,22 @@ export function expandToken(token, opts, platform) {
       expandedTokenObjRef = expandedTokenObjRef[index + 1];
     }
     Object.entries(objectVal).forEach(([key, value]) => {
-      expandedTokenObjRef[key] = {
-        ...copyMeta,
+      expandedTokenObjRef[key] = deepmerge(copyMeta, {
         [`${uses$ ? '$' : ''}value`]: value,
         [`${uses$ ? '$' : ''}type`]: getTypeFromMap(key, compositionType, typesMap),
-      };
+        // store the original composite token type as well as which prop it was
+        // so that for example a lineHeight property of a typography token, when expanded
+        // can still be matched as a lineHeight token, even though its type is now "number" after the expansion
+        $extensions: {
+          'com.styledictionary': {
+            preExpand: {
+              type: compositionType,
```

Comment:
Should this be `compositionType` to be more explicit?

Reply:
`compositeType` I guess? Although both is probably fine